docs/ndcube.rst

```diff
@@ -0,0 +1,355 @@
+  >>> my_cube = ndcube.NDCube(data, input_wcs, uncertainty=np.sqrt(data),
+  ...                         mask=mask, meta=meta, unit=None) 
+
+
+Dimensions
+----------
+
+NDCube has a useful property for inspecting its data shape and
+axis types, `~ndcube.NDCube.dimensions`::
+
+  >>> my_cube.dimensions
+  DimensionPair(shape=<Quantity [ 3., 4., 5.] pix>, axis_types=['HPLN-TAN', 'HPLT-TAN', 'WAVE'])
+
+This returns a named tuple with a "shape" and "axis_types" attribute.
+"shape" is an `~astropy.units.Quantity` of pixel units giving the
+length of each dimension in the `~ndcube.NDCube`.  Meanwhile,
+"axis_types" is `list` of `str` giving the WCS transformation type for
+each axis. Here the shape and axis types are given in data order, not
+WCS order.
+
+As the dimensions property returns a named tuple, the shape and axis
+types can be accessed directly::
+
+  >>> my_cube.dimensions.shape
+  <Quantity [ 3., 4., 5.] pix>
+  >>> my_cube.dimensions.axis_types
+  ['HPLN-TAN', 'HPLT-TAN', 'WAVE']
+  
+Slicing
+-------
+
+Arguably NDCube's most powerful capability is its slicing.  Slicing an
+NDCube object using the standard slicing notation allows users to
+access sub-regions of their data while simultaneously slicing not only
+the other array attributes (e.g. uncertainty, mask, etc.) but also the
+WCS object.  This ensures that even though the data array has changed
+size and shape, each array element will still corresponding to the
+same real world coordinates as they did before.  An example of how to
+slice a 3-D `~ndcube.NDCube` object is::
+
+  >>> my_cube_roi = my_cube[3:5, 10:100, 30:37]
+
+Slicing can also reduce the dimension of an `~ndcube.NDCube`, e.g.::
+
+  >>> my_2d_cube = my_cube[0, 10:100, 30:37]
+
+In addition to slicing by index, `~ndcube.NDCube` supports a basic
+version of slicing/index by real world coordinate via the
+`~ndcube.NDCube.crop_by_coords` method.  This takes a list of
+`astropy.units.Quantity` representing the real world coordinates of
+the lower left corner of the region of interest.  The order of the
+coordinates must be the same as the order of the data axes.  A second
+iterable of `~astropy.units.Quantity` must also be provided which gives
+the widths of the region of interest in each data axis::
+
+  >>> from astropy.units import Quantity
+  >>> my_cube_roi = my_cube.crop_by_coords(
+  ... [Quantity(0.7, unit="deg"), Quantity(1.3e-5, unit="deg"), Quantity(1.04e-9, unit="m")],
+  ... [Quantity(0.6, unit="deg"), Quantity(1., unit="deg"), Quantity(0.08e-9, unit="m")])
+
+This method does not rebin or interpolate the data if the region of interest
+defined does not perfectly map onto the array's "pixel" grid.  Instead
+it translates from real world to pixel coordinates and rounds to the
+nearest integer before indexing/slicing the `~ndcube.NDCube` object.
+Therefore it should be noted that slightly different inputs to this
+method can result in the same output.
```

Comment:
Is it also worth commenting here that it will crop a square box in pixel coordinates that will be larger if the data is rotated?

Reply:
Will it?  Why?  Assuming you're right, perhaps you would be a better person to explain that?

Reply:
Assuming it's the same as the map implementation it should calculate the corner pixel locations and then crop based on the extrema of them.

Reply:
Oh I see what you're saying.  I think it's the same as map...